src/app/components/auth/login/login.component.spec.ts

```diff
@@ -36,10 +56,43 @@ describe('LoginComponent', () => {
   beforeEach(() => {
     fixture = TestBed.createComponent(LoginComponent);
     component = fixture.componentInstance;
-    fixture.detectChanges();
+    apiService = TestBed.get(ApiService);
   });
 
   it('should create', () => {
+    fixture.detectChanges();
     expect(component).toBeTruthy();
   });
+
+  it('should call userLogin Successfully', () => {
+    spyOn(apiService, 'postUrl').and.returnValue(new Observable((observer) => {
+      observer.next({'token': [{}]});
+      observer.complete();
+      return {unsubscribe() {}};
+    }));
+    fixture.detectChanges();
+    component.userLogin(false);
+    component.userLogin(true);
+    expect(apiService.postUrl).toHaveBeenCalled();
+  });
+
+  it('should call userLogin UnSuccessfully', () => {
+    let err = {};
+    spyOn(apiService, 'postUrl').and.returnValue(new Observable((observer) => {
+      observer.error(err);
+      return {unsubscribe() {}};
+    }));
+    fixture.detectChanges();
+    component.userLogin(true);
+
+    err =  {'status': 400};
```

Comment:
I think these 3 lines are not needed.

Reply:
These are required to call catch statement of the try and catch block